### DISCUSSION TITLE
Open file without leaving neo-tree

### DISCUSSION THREAD

**Discussion by dpetka2001** [nvim-neo-tree/neo-tree.nvim#919] [General]
As the title says, is it possible to open a file from neo-tree and keep the focus in neo-tree buffer? Currently when i open a file it changes focus to that buffer.

**Comment by pysan3** [comment]
No, by default.

Neotree uses a simple `:e <file-name>` when opening a file, so it's simply a vim feature to focus the opened file.

However, if you use a callback, you would be able to achieve what you want.
This will make you focus neo-tree even after a file is opened.

```lua
    event_handlers = {
      {
        event = "file_opened",
        handler = function(_)
          vim.cmd([[Neotree focus]])
        end,
      },
    }
```

Or another option is to make a new keybind since with the above callback, you'll never be able to escape neo-tree easily.

```lua
    filesystem = {
      window = {
        mappings = {
          ["L"] = "open_nofocus",
        },
      },
      commands = {
        open_nofocus = function(state)
          require("neo-tree.sources.filesystem.commands").open(state)
          vim.schedule(function()
            vim.cmd([[Neotree focus]])
          end)
        end,
      },
    },
```

**Reply by dpetka2001** [reply]
Thank you for your answer, I will try to play around with these options during the weekend when I have some more free time.

**Reply by dpetka2001** [reply]
I managed to try it right now and your second suggestion with new keybind works wonders just how I wanted it. Thank you again for your help. :smile:
